src/core_plugins/kibana/public/dashboard/dashboard_app.js

```diff
@@ -1,24 +1,21 @@
 import _ from 'lodash';
 import angular from 'angular';
 import { uiModules } from 'ui/modules';
-import uiRoutes from 'ui/routes';
 import chrome from 'ui/chrome';
 
 import 'ui/query_bar';
 
-import { SavedObjectNotFound } from 'ui/errors';
 import { getDashboardTitle, getUnsavedChangesWarningMessage } from './dashboard_strings';
 import { DashboardViewMode } from './dashboard_view_mode';
 import { TopNavIds } from './top_nav/top_nav_ids';
 import { ConfirmationButtonTypes } from 'ui/modals/confirm_modal';
-import dashboardTemplate from 'plugins/kibana/dashboard/dashboard.html';
 import { FilterBarQueryFilterProvider } from 'ui/filter_bar/query_filter';
 import { DocTitleProvider } from 'ui/doc_title';
 import { getTopNavConfig } from './top_nav/get_top_nav_config';
 import { DashboardConstants, createDashboardEditUrl } from './dashboard_constants';
 import { VisualizeConstants } from 'plugins/kibana/visualize/visualize_constants';
-import { DashboardState } from './dashboard_state';
-import { notify } from 'ui/notify';
+import { DashboardStateManager } from './dashboard_state_manager';
```

Comment:
Just in general (and _not_ something you have to change in this PR): As this PR is touching a lot of code it would simplify the review if this renaming was in a separate PR (as it adds "mental overhead" to the process).

Reply:
Yeaa, it's a good point to raise, and something I did consider.  I actually held off on some other renamings I want to do (e.g. panelIndex => panelId and id => savedObjectId).  It was a difficult decision, but I ended up doing the renaming because I thought it would help clarify the new role that class is playing, and `dashboard_state` kinda conflicted with the new redux state tree.  But good to bring up, I will be sure to try to cut back on any unnecessary renames in big PRs like this! 👍

Reply:
:+1: